composables/helpers/buildOrdersFilterInput.ts

```diff
@@ -6,12 +7,29 @@ export const buildOrdersFilterInput = (params: {
   const variables = {
     filter: "",
     startIndex: 0,
-    pageSize: 0,
+    pageSize: 20,
+  }
+
+  const searchFilters = ref([])
+
+  const getPastDateTimestamp = (months) => {
+    const today = new Date()
+    today.setMonth(today.getMonth() - months)
+    return today.toISOString()
   }
 
   if (params.filters) {
-    // @TODO logic
-    variables.filter = ""
+    for (const filters of params.filters) {
+      const filter = filters.split("-")
+      if (filter[0] === "M") {
+        searchFilters.value.push(`createDate gt ${getPastDateTimestamp(parseInt(filter[1]))}`)
+      } else if (filter[0] === "Y") {
+        const startDate = new Date(`${filter[1]}-01-01`).toISOString()
+        const endDate = new Date(`${filter[1]}-12-31`).toISOString()
+        searchFilters.value.push(`createDate gt ${startDate} and createDate lt ${endDate}`)
+      }
+    }
```

Comment:
As a best practice always convert string to lowercase and then compare.
ex. if (filter[0].toLowerCase() === "m")

Reply:
Resolved